velox/common/memory/MallocAllocator.cpp

```diff
@@ -65,30 +65,43 @@ bool MallocAllocator::allocateNonContiguousWithoutRetry(
 
   std::vector<void*> pages;
   pages.reserve(mix.numSizes);
+  bool failedToAllocate = false;
   for (int32_t i = 0; i < mix.numSizes; ++i) {
     // Trigger allocation failure by breaking out the loop.
     if (testingHasInjectedFailure(InjectedFailure::kAllocate)) {
+      failedToAllocate = true;
       break;
     }
-    MachinePageCount numSizeClassPages =
-        mix.sizeCounts[i] * sizeClassSizes_[mix.sizeIndices[i]];
-    void* ptr;
-    stats_.recordAllocate(
-        AllocationTraits::pageBytes(sizeClassSizes_[mix.sizeIndices[i]]),
-        mix.sizeCounts[i],
-        [&]() {
-          ptr = ::malloc(
-              AllocationTraits::pageBytes(numSizeClassPages)); // NOLINT
-        });
-    if (ptr == nullptr) {
-      // Failed to allocate memory from memory.
-      break;
+
+    const auto maxClassPagesInRun = static_cast<int>(
+        Allocation::PageRun::kMaxPagesInRun /
+        sizeClassSizes_[mix.sizeIndices[i]]);
```

Comment:
Can we initialize sizeClassMaxUnits_ once in construction in MemoryAllocator base class? Thanks!